pkg/frontend/admin_openshiftcluster_redeployvm.go

```diff
@@ -32,8 +32,17 @@ func (f *frontend) postAdminOpenShiftClusterRedeployVM(w http.ResponseWriter, r
 func (f *frontend) _postAdminOpenShiftClusterRedeployVM(ctx context.Context, r *http.Request) error {
 	vars := mux.Vars(r)
 
+	subscriptionDoc, err := f.db.Subscriptions.Get(ctx, vars["subscriptionId"])
+	if err != nil {
+		return err
+	}
+
+	if subscriptionDoc == nil {
+		return api.NewCloudError(http.StatusNotFound, api.CloudErrorCodeSubscriptionNotFound, "", "The subscription '%s' was not found.", vars["subscriptionId"])
+	}
```

Comment:
this is duplicated in `func (f *frontend) validateSubscriptionState` - could you split that function into two and consolidate the frontend changes in this PR?